bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -196,6 +201,19 @@ protected void triggerChannelIfRequired(String channelId) {
         return detectedObjectTypes;
     }
 
+    private static Optional<NAWelcomeSubEvent.TypeEnum> findDetectedCategory(NAWelcomeEvent event) {
+        String category = event.getCategory();
+        if (category != null) {
+            switch (category) {
+                case "human":
```

Comment:
Couldn't you use NAWelcomeSubEvent.TypeEnum.valueOf(catory.uppercase()) instead ?

Reply:
Hm, that would result in an IllegalArgumentException when Netatmo returns a category which isn't existing within the enum.

Reply:
According to [documentation](https://dev.netatmo.com/apidocumentation/security) the set of values seems clearly defined, so, don't you think we could rely on it ? We'll face a potential IAE if the API evolves...

Reply:
You have found a documentation of the possible values of the "category" element? :-) I haven't found any documentation regarding this element, so we don't know which value could be set.

Reply:
Yes, take a look at 'event_list' on the referenced doc url.

Reply:
But the "event_list" structure or outdoor events have nothing to do with the undocumented "category" element.

I saw the same values ("human" and "animal"), but there is no guarantuee that there are not more possible values, the element and it's possible values aren't documented...

So it is a risk to use the same enum and could also today lead to an IllegalArgumentException.

If you say "yes" I change it to use the enum. ;-) It is some time left since 2.5.8, so potential errors could occur before to be able to fix that again.

Reply:
Ah, it was answered by Netatmo yesterday, it should be safe to use the enum. :-)

https://forum.netatmo.com/viewtopic.php?f=5&t=18506

Reply:
So, I think it is safe to proceed with an enum !

Reply:
An unexpected value in the API should not result in an unhandled exception. I'd go for the enum version, too. Although it's not nice, you could catch the IAE.

Reply:
It is now declared as an enum within the swagger spec, like the sub-event type.